scripts/modules/ai-services-unified.js

```diff
@@ -661,11 +663,21 @@ async function _unifiedServiceRunner(serviceType, params) {
 				throw new Error('User prompt content is missing.');
 			}
 
+			// Some CLI-backed providers do not support temperature
+			const lowerProvider = providerName?.toLowerCase();
+			const providerSupportsTemperature = ![
+				'codex-cli',
+				'claude-code'
+			].includes(lowerProvider);
+
 			const callParams = {
 				apiKey,
 				modelId,
 				maxTokens: roleParams.maxTokens,
-				temperature: roleParams.temperature,
+				...(providerSupportsTemperature &&
+				typeof roleParams.temperature === 'number'
+					? { temperature: roleParams.temperature }
+					: {}),
```

Comment:
lets make sure this type of logic is handled inside the provider ? maybe add to the baseProvider ? and then apply inside the class of the children ? trying to reduce the logic inside this module as much as I can